src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -116,101 +116,106 @@ export class NodeClientRequest extends ClientRequest {
     this.observer.emit('request', isomorphicRequest)
 
     this.log('executing response resolver...')
-    const [resolverError, mockedResponse] = await until(async () =>
-      this.resolver(isomorphicRequest, this.response)
-    )
-
-    // Halt the request whenever the resolver throws an exception.
-    if (resolverError) {
-      this.log(
-        'encountered resolver exception, aborting request...',
-        resolverError
-      )
-      this.emit('error', resolverError)
-      this.terminate()
-
-      return
-    }
-
-    if (mockedResponse) {
-      this.log('received mocked response:', mockedResponse)
-      this.responseSource = 'mock'
-
-      const isomorphicResponse = toIsoResponse(mockedResponse)
-      this.respondWith(mockedResponse)
-      this.log(
-        isomorphicResponse.status,
-        isomorphicResponse.statusText,
-        isomorphicResponse.body,
-        '(MOCKED)'
-      )
 
-      callback?.()
-
-      this.log('emitting the custom "response" event...')
-      this.observer.emit('response', isomorphicRequest, isomorphicResponse)
+    // Execute the resolver Promise like a side-effect.
+    // Node.js 16 forces "ClientRequest.end" to be synchronous and return "this".
+    until(async () => this.resolver(isomorphicRequest, this.response)).then(
```

Comment:
Resolver execution is a side-effect right now. This needs to be tested.

Reply:
Automated tests report no issues. Response handling logic is asynchronous by design. I think when Node 16 forced `end` to return `this` they accounted for it. As in: return the next (potentially unresolved) state of `ClientRequest`.